File: SourceCode/Debug Helpers/open.fda.gov/ApiDalc/DataObjects/ResultRecall.vb
#Region " Imports "

Imports Newtonsoft.Json.Linq ' OpenSource

#End Region

Namespace DataObjects

    ''' <summary>
    ''' Recall Result
    ''' </summary>
    ''' <remarks></remarks>
    Public Class ResultRecall

#Region " Public Properties "

        Public Property KeyWord As String = String.Empty

        Public Property City As String
        Public Property Classification As String
        Public Property Code_info As String
        Public Property Country As String
        Public Property Distribution_Pattern As String
        Public Property Event_Id As String
        Public Property Initial_Firm_Notification As String
        Public Property Openfda As Object
        Public Property Product_Description As String
        Public Property Product_Quantity As String
        Public Property Product_Type As String
        Public Property Reason_For_Recall As String
        Public Property Recall_Initiation_Date As String
        Public Property Recall_Number As String
        Public Property Recalling_Firm As String
        Public Property Report_Date As String
        Public Property State As String
        Public Property Status As String
        Public Property Voluntary_Mandated As String

#End Region

#Region " Public Methods "

        ''' <summary>
        ''' Converts JSON Data to List
        ''' </summary>
        ''' <param name="jsondata">JSON Object Data</param>
        ''' <returns>List of Recall Data</returns>
        ''' <remarks></remarks>
        Public Shared Function CnvJsonDataToList(jsondata As JObject) As List(Of ResultRecall)

            Dim result As New List(Of ResultRecall)

            For Each obj In jsondata.GetValue("results")

                Dim tmp As New ResultRecall

                tmp.City = CStr(obj("city"))
                tmp.Classification = CStr(obj("classification"))
                tmp.Code_info = CStr(obj("code_info"))
                tmp.Country = CStr(obj("country"))
                tmp.Distribution_Pattern = CStr(obj("distribution_pattern"))
                tmp.Event_Id = CStr(obj("event_id"))
                tmp.Initial_Firm_Notification = CStr(obj("initial_firm_notification"))
                'tmp.openfda = zz("city")
                tmp.Product_Description = CStr(obj("product_description"))
                tmp.Product_Quantity = CStr(obj("product_quantity"))
                tmp.Product_Type = CStr(obj("product_type"))
                tmp.Reason_For_Recall = CStr(obj("reason_for_recall"))
                tmp.Recall_Initiation_Date = CStr(obj("recall_initiation_date"))
                tmp.Recall_Number = CStr(obj("recall_number"))
                tmp.Recalling_Firm = CStr(obj("recalling_firm"))
                tmp.Report_Date = CStr(obj("report_date"))
                tmp.State = CStr(obj("state"))
                tmp.Status = CStr(obj("status"))
                tmp.Voluntary_Mandated = CStr(obj("voluntary_mandated"))

                result.Add(tmp)

            Next

            Return result

        End Function

        ''' <summary>
        ''' Converts JSON Data to List
        ''' </summary>
        ''' <param name="jsondata">JSON String Data</param>
        ''' <returns>List of Recall Data</returns>
        ''' <remarks></remarks>
        Public Shared Function CnvJsonDataToList(jsondata As String) As List(Of ResultRecall)

            Dim jo As JObject = JObject.Parse(jsondata)

            Return CnvJsonDataToList(jo)

        End Function

        Public Overrides Function ToString() As String
            Return String.Format("{0} - {1}", Recall_Initiation_Date, Classification)
        End Function

#End Region

    End Class

End Namespace

File: SourceCode/AspCoreVersion/src/ShopAware.Core/DataObjects/ResultRecall.cs
using System.Collections.Generic;
using Newtonsoft.Json.Linq;

namespace ShopAware.Core
{
    namespace DataObjects
    {
        /// <summary>
        ///     Recall Result
        /// </summary>
        /// <remarks></remarks>
        public class ResultRecall
        {
            #region Member Variables

            private string _KeyWord = string.Empty;

            #endregion

            #region Properties

            public string KeyWord
            {
                get
                {
                    return _KeyWord;
                }
                set
                {
                    _KeyWord = value;
                }
            }

            public string City { get; set; }

            public string Classification { get; set; }

            public string Code_info { get; set; }

            public string Country { get; set; }

            public string Distribution_Pattern { get; set; }

            public string Event_Id { get; set; }

            public string Initial_Firm_Notification { get; set; }

            public object Openfda { get; set; }

            public string Product_Description { get; set; }

            public string Product_Quantity { get; set; }

            public string Product_Type { get; set; }

            public string Reason_For_Recall { get; set; }

            public string Recall_Initiation_Date { get; set; }

            public string Recall_Number { get; set; }

            public string Recalling_Firm { get; set; }

            public string Report_Date { get; set; }

            public string State { get; set; }

            public string Status { get; set; }

            public string Voluntary_Mandated { get; set; }

            #endregion

            #region Public Methods

            /// <summary>
            ///     Converts JSON Data to List
            /// </summary>
            /// <param name="jsondata">JSON Object Data</param>
            /// <returns>List of Recall Data</returns>
            /// <remarks></remarks>
            public static List<ResultRecall> CnvJsonDataToList(JObject jsondata)
            {
                var result = new List<ResultRecall>();

                foreach (var obj in jsondata.GetValue("results"))
                {
                    var tmp = new ResultRecall();

                    tmp.City = (obj["city"]).ToString();
                    tmp.Classification = (obj["classification"]).ToString();
                    tmp.Code_info = (obj["code_info"]).ToString();
                    tmp.Country = (obj["country"]).ToString();
                    tmp.Distribution_Pattern = (obj["distribution_pattern"]).ToString();
                    tmp.Event_Id = (obj["event_id"]).ToString();
                    tmp.Initial_Firm_Notification = (obj["initial_firm_notification"]).ToString();
                    //tmp.openfda = zz("city")
                    tmp.Product_Description = (obj["product_description"]).ToString();
                    tmp.Product_Quantity = (obj["product_quantity"]).ToString();
                    tmp.Product_Type = (obj["product_type"]).ToString();
                    tmp.Reason_For_Recall = (obj["reason_for_recall"]).ToString();
                    tmp.Recall_Initiation_Date = (obj["recall_initiation_date"]).ToString();
                    tmp.Recall_Number = (obj["recall_number"]).ToString();
                    tmp.Recalling_Firm = (obj["recalling_firm"]).ToString();
                    tmp.Report_Date = (obj["report_date"]).ToString();
                    tmp.State = (obj["state"]).ToString();
                    tmp.Status = (obj["status"]).ToString();
                    tmp.Voluntary_Mandated = (obj["voluntary_mandated"]).ToString();

                    result.Add(tmp);
                }

                return result;
            }

            /// <summary>
            ///     Converts JSON Data to List
            /// </summary>
            /// <param name="jsondata">JSON String Data</param>
            /// <returns>List of Recall Data</returns>
            /// <remarks></remarks>
            public static List<ResultRecall> CnvJsonDataToList(string jsondata)
            {
                var jo = JObject.Parse(jsondata);

                return CnvJsonDataToList(jo);
            }

            public override string ToString()
            {
                return string.Format("{0} - {1}", Recall_Initiation_Date, Classification);
            }

            #endregion
        }
    }
}